client/reader/stream/post-unavailable.jsx

```diff
@@ -4,26 +4,20 @@ import { localize } from 'i18n-calypso';
 import { PureComponent } from 'react';
 
 class PostUnavailable extends PureComponent {
-	componentDidMount() {
-		this.errors = {
-			unauthorized: this.props.translate(
-				'This is a post on a private site that you’re following, but not currently a member of.' +
-					' ' +
-					'Please request membership to display these posts in Reader.'
-			),
-			default: this.props.translate( 'An error occurred loading this post.' ),
-		};
-	}
-
 	render() {
-		const errorMessage =
-			this.errors[ this.props.post.errorCode || 'default' ] || this.errors.default;
+		const error = this.props.post.error;
+
+		if ( ! error ) {
+			return null;
+		}
 
-		if ( this.props.post.statusCode === 404 ) {
+		if ( error.status === 404 || error.error === 404 ) {
```

Comment:
Strangely, the errors I'm seeing have `error.status` as `400` and `error.error` as `404`... I'm not sure why.

Reply:
yea, that is strange.

Reply:
Ahh I see that the call /read/sites/150893645/posts/8094111 actually returned with http status `200` and body `{"code":400,"body":{"error":404,"message":"Post not found"}}`, so that's just what the API is returning. I wish we had have designed it to use error http status codes ratther than wrapping everything in `200 OK`. That way we could easily see the errors in the dev tools network panel and error handling would work as expected by default. 🤷